packages/eslint-plugin/src/rules/require-array-sort-compare.ts

```diff
@@ -66,23 +66,26 @@ export default createRule<Options, MessageIds>({
       return false;
     }
 
-    return {
-      "CallExpression[arguments.length=0] > MemberExpression[property.name='sort'][computed=false]"(
-        callee: TSESTree.MemberExpression,
-      ): void {
-        const calleeObjType = getConstrainedTypeAtLocation(
-          services,
-          callee.object,
-        );
+    function checkSortArgument(callee: TSESTree.MemberExpression): void {
+      const calleeObjType = getConstrainedTypeAtLocation(
+        services,
+        callee.object,
+      );
 
-        if (options.ignoreStringArrays && isStringArrayNode(callee.object)) {
-          return;
-        }
+      if (options.ignoreStringArrays && isStringArrayNode(callee.object)) {
+        return;
+      }
 
-        if (isTypeArrayTypeOrUnionOfArrayTypes(calleeObjType, checker)) {
-          context.report({ node: callee.parent, messageId: 'requireCompare' });
-        }
-      },
+      if (isTypeArrayTypeOrUnionOfArrayTypes(calleeObjType, checker)) {
+        context.report({ node: callee.parent, messageId: 'requireCompare' });
+      }
+    }
+
+    return {
+      "CallExpression[arguments.length=0] > MemberExpression[property.name='sort'][computed=false]":
+        checkSortArgument,
+      "CallExpression[arguments.length=0] > MemberExpression[property.name='toSorted'][computed=false]":
```

Comment:
[Non-Actionable] Aside: I'm glad you didn't increase the complexity of this selector 😂 I think it's at the maximum reasonable level to still be readable as-is...

Reply:
lol, that was because i simply do not know the select syntax well enough to even try